### PR TITLE
enable cuda build

### PR DESCRIPTION
**Pull request by xiki-tempula**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xiki-tempula** [comment]
@conda-forge-admin, please rerender

**Comment by mattwthompson** [comment]
#119

**Comment by xiki-tempula** [comment]
The CUDA builds but the MPI+cuda failed due to `No space left on device`

**Comment by xiki-tempula** [comment]
@conda-forge-admin, please rerender

**Comment by xiki-tempula** [comment]
@mattwthompson This working fine when tested locally. Shall we merge this PR?

**Review by mattwthompson — COMMENTED** [review]
The green checks are good and the changes pass the eye test, but I'd like one more person to either skim the changes or test them locally.

Are there any executables that can be turned on in this file with the inclusion of CUDA?

https://github.com/conda-forge/ambertools-feedstock/blob/main/recipe/run_test.sh

**Comment by xiki-tempula** [comment]
@mattwthompson I think we could potentially test `cpptraj.cuda`. However, I don't know how to test this only when there is `cuda`. I wonder if I can have some of your help? Thank you.

**Comment by mattwthompson** [comment]
I think it can be tucked away under a conditional like already happens for MPI: https://github.com/conda-forge/ambertools-feedstock/blob/bc93c747c80f8592d002ea0aa5ac321a82e9fe6f/recipe/run_test.sh#L50-L56

**Comment by xiki-tempula** [comment]
@conda-forge-admin, please rerender

**Comment by xiki-tempula** [comment]
@mattwthompson Ok, I have added a `which` and a `lld` test for `cpptraj.cuda`, which seems to pass.

**Comment by xiki-tempula** [comment]
@conda-forge-admin, please restart ci​

**Comment by xiki-tempula** [comment]
@conda-forge-admin, please rerender

**Comment by xiki-tempula** [comment]
@conda-forge-admin, please restart ci​

**Comment by xiki-tempula** [comment]
@mattwthompson Shall we merge this before working on 24? All the builds are passing and I have test for cpptraj.cuda.

**Comment by mattwthompson** [comment]
It's probably fine but I'd like somebody from the Amber community to chime in

**Comment by xiki-tempula** [comment]
@njzjz Do you mind give this PR a review? Thanks.

**Comment by mattwthompson** [comment]
You can re-configure the build process to store artifacts if they're not already there (I haven't checked), re-render, and then on the next build they're be available to download somewhere on the Azure pages that are linked in "CI" from this PR. Then you can install those as local conda packages

https://conda-forge.org/docs/maintainer/conda_forge_yml/#azure

Everything can be done locally (there's a `build_locally.py` file somewhere and/or information in the docs https://conda-forge.org/docs/maintainer/) but given the complexity of this feedstock I would worry there's a risk of false positives compared to what happens in automation

**Review by njzjz — APPROVED** [review]

**Comment by xiki-tempula** [comment]
@dacase Building the linux version is quite easy, you just find a linux machine. clone the repo, run `build_locally.py` and select the relevant option. Then upload the `.conda` file to your local conda channel and you can then install it with conda.
In our company, we use this branch to build amber, where I change the source to build from local tar.gz file. Then upload the `.conda` to company conda channel and deploy in our production environment where `pmemd.cuda.mpi` works as expected.

**Comment by mattwthompson** [comment]
@xiki-tempula could you merge the `main` branch into this once more? I don't want to accidentally revert #147

After that I'll merge tomorrow morning if nobody else got to it by then (or objects)

File: .ci_support/linux_64_c_compiler_version11c_stdlib_version2.17cuda_compilernvcccuda_compiler_version11.8cxx_compiler_version11fortran_compiler_version11mpimpichnumpy1.22python3.10.____cpython.yaml
Comment (xiki-tempula) [inline .ci_support/linux_64_c_compiler_version11c_stdlib_version2.17cuda_compilernvcccuda_compiler_version11.8cxx_compiler_version11fortran_compiler_version11mpimpichnumpy1.22python3.10.____cpython.yaml:52]: @mattwthompson I have merged the main to this branch and rerendered the branch. You could see that openmpi is version 5 and zlib is 1

**Pull request merged**
